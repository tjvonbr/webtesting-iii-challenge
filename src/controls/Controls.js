import React from 'react';

const Controls = props => {
  const { locked, closed, toggleLocked, toggleClosed } = props;

  return (
    <div className="controls panel">
      <button disabled={!closed} 
              onClick={toggleLocked} 
              className="toggle-btn"
              data-testid="lock-btn">
        {locked ? 'Unlock Gate' : 'Lock Gate'}
      </button>
      <button disabled={locked} 
              onClick={toggleClosed} 
              className="toggle-btn"
              data-testid="open-btn">
        {closed ? 'Open Gate' : 'Close Gate'}
      </button>
    </div>
  );
};

export default Controls;
