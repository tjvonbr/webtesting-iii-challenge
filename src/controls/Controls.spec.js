import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, render, fireEvent } from '@testing-library/react';

import Controls from './Controls.js';

afterEach(cleanup);

describe('<Controls />', () => {
  it('should match new snapshot', () => {
    const tree = renderer
      .create(<Controls />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('toggleLocked function', () => {
  it('locks the gate if unlocked and unlocks the gate if locked', () => {
    const toggleLocked = jest.fn();

    const { getByTestId } = render(<Controls closed={true} locked={true} />);

    const openButton = getByTestId('open-btn');
    const unlockButton = getByTestId('lock-btn');

    fireEvent.click(openButton);

    expect(toggleLocked).toHaveBeenCalled();

    console.log(openButton);
  });
});