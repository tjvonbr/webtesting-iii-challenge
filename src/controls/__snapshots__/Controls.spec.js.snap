// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls /> should match new snapshot 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    data-testid="lock-btn"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
    data-testid="open-btn"
  >
    Close Gate
  </button>
</div>
`;
